File: dispositivi-service/src/main/java/com/uniupo/dispositivi/model/Totem.java
package com.uniupo.dispositivi.model;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;

@Entity
@DiscriminatorValue("TOTEM")
public class Totem extends Dispositivo {

    public Totem() {
        super();
        this.tipoDispositivo = "TOTEM";
    }

    public Totem(Boolean status, Integer corsia, Integer casello) {
        super(status, corsia, casello, "TOTEM");
    }
}

File: IOT/ESP32_Backup/Totem.py
class Dispositivo:
    def __init__(self, status=False, corsia=0, casello=0, tipo_dispositivo="", dispositivo_id=None):
        self._id = dispositivo_id  
        self.corsia = corsia
        self.casello = casello
        self.status = status
        self.tipo_dispositivo = tipo_dispositivo